imagebuildah/stage_executor.go

```diff
@@ -362,6 +365,28 @@ func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) err
 		stripSetgid := false
 		preserveOwnership := false
 		contextDir := s.executor.contextDir
+
+		// If we are copying files via heredoc syntax, then
+		// its time to create these temporary files on host
+		// and copy these to container
+		if len(copy.Files) > 0 {
+			copySources := []string{}
+			for _, file := range copy.Files {
+				tmpFile, err := os.Create(file.Name)
```

Comment:
Writing it in `/var/tmp` on the host and then cleaning it up, i am wonder if there is any need to further sanitize the `file.Name` cause heredoc parser rejects funny strings for heredoc names like ones with `..` or `/` .

> But then if we're only creating them so that they can be copied into the rootfs, it might be preferable to generate an archive stream with archive/tar and extract that stream into the rootfs with copier.Put().

`s.builder.Add(` takes path as sources and does all the work related to perms and bit options if i'm gonna use `copier.Put` here, we will have to re-implement the logic or share it with `buildah.Add` therefore I thought using the same logic is more convenient, therefore creating a file and then cleaning it just after `copy` process.

Reply:
> Writing it in `/var/tmp` on the host and then cleaning it up, i am wonder if there is any need to further sanitize the `file.Name` cause heredoc parser rejects funny strings for heredoc names like ones with `..` or `/` .

path.Base(filepath.ToSlash(file.Name)), then.  We can't predict if that parser will become more forgiving later.

> > But then if we're only creating them so that they can be copied into the rootfs, it might be preferable to generate an archive stream with archive/tar and extract that stream into the rootfs with copier.Put().
> 
> `s.builder.Add(` takes path as sources and does all the work related to perms and bit options if i'm gonna use `copier.Put` here, we will have to re-implement the logic or share it with `buildah.Add` therefore I thought using the same logic is more convenient, therefore creating a file and then cleaning it just after `copy` process.

What are the expected permissions on a COPY heredoc?  0711 seems like an odd choice to hard wire, since that's execute for the owning group and everyone, but read-write-execute for the owner.

Reply:
> path.Base(filepath.ToSlash(file.Name)), then. We can't predict if that parser will become more forgiving later.
I added this change, but can add `copier.Put` if that is better.